src/config/mongoose_config_validator_toml.erl

```diff
@@ -452,4 +1584,225 @@ validate_pool_scope(Value) -> validate_enum(Value, [host, global]).
+
+validate_ipv6_mask(Mask) ->
+    validate_range(Mask, 0, 128).
+
+validate_network_address(Value) ->
+    ?LOG_DEBUG(#{what => validate_network_address,
+                 value => Value}),
+    validate_oneof(Value, [fun validate_domain/1, fun validate_ip_address/1]).
+
+validate_oneof(Value, Funs) ->
+    Results = [safe_call_validator(F, Value) || F <- Funs],
+    case lists:any(fun(R) -> R =:= ok end, Results) of
+        true ->
+            ok;
+        false ->
+            error(#{what => validate_oneof_failed,
+                    validation_results => Results})
+    end.
+
+safe_call_validator(F, Value) ->
+    try
+        F(Value),
+        ok
+    catch error:Reason:Stacktrace ->
+              #{reason => Reason, stacktrace => Stacktrace}
+    end.
+
+validate_network_port(Value) ->
+    validate_range(Value, 0, 65535).
+
+validate_range(Value, Min, Max) when Value >= Min, Value =< Max ->
+    ok.
+
+validate_wpool_strategy(Value) ->
+    validate_enum(Value, [best_worker, random_worker, next_worker,
+                          available_worker, next_available_worker]).
+
+validate_filename(Filename) ->
+    case file:read_file_info(Filename) of
+        {ok, _} ->
+            ok;
+        Reason ->
+            error(#{what => invalid_filename, filename => Filename, reason => Reason})
+    end.
+
+validate_dirname(Dirname) ->
+    case file:list_dir(Dirname) of
+        {ok, _} ->
+            ok;
+        Reason ->
+            error(#{what => invalid_dirname, dirname => Dirname, reason => Reason})
+    end.
+
+validate_keystore_key({Name, ram}) ->
+    validate_non_empty_atom(Name);
+validate_keystore_key({Name, {file, Path}}) ->
+    validate_non_empty_atom(Name),
+    validate_filename(Path).
+
+validate_muc_affiliation_rule({{User, Server, Resource}, Affiliation}) ->
+    validate_non_empty_binary(User),
+    validate_binary_domain(Server),
+    validate_binary(Resource),
+    validate_non_empty_atom(Affiliation).
+
+validate_maybe_css_file(false) ->
+    ok;
+validate_maybe_css_file(Bin) ->
+    validate_non_empty_binary(Bin). %% Could be more precise type
+
+validate_top_link({Url, Text}) ->
+    validate_url(Url),
+    validate_non_empty_string(Text).
+
+validate_muc_config_schema({Field, Value}) ->
+    validate_non_empty_string(Field),
+    validate_string(Value);
+validate_muc_config_schema({Field, Value, InternalField, FieldType})
+    when is_list(Value); is_float(Value); is_integer(Value) ->
+    validate_non_empty_string(Field),
+    validate_enum(FieldType, [binary, integer, float]),
+    validate_non_empty_atom(InternalField).
+
+validate_pubsub_nodetree(Value) ->
+    validate_non_empty_binary(Value),
+    validate_backend(nodetree, b2a(Value)).
+
+validate_pubsub_plugin(Value) ->
+    validate_non_empty_binary(Value),
+    validate_backend(node, b2a(Value)).
+
+validate_pubsub_pep_mapping({Namespace, Id}) ->
+    validate_non_empty_string(Namespace),
+    validate_non_empty_string(Id).
```

Comment:
Maybe I am nitpicking now, but wouldn't `Node` variable name be more convenient, as it would match corresponding key in struct? See `config_parser_SUITE.erl:2298,2390,2391`, as direct links to the lines of files that "are not rendered by default" doesn't make github come up with their rendering 🙄